substrate/frame/bags-list/src/benchmarks.rs

```diff
@@ -348,9 +348,72 @@ benchmarks_instance_pallet! {
 		)
 	}
 
+	on_idle {
+		// This benchmark generates weights for `on_idle` based on runtime configuration.
+		// The main input is the runtime's `MaxAutoRebagPerBlock` type, which defines how many
+		// nodes can be rebagged per block.
+		// This benchmark simulates a more realistic and fragmented rebag scenario.
+
+		List::<T, _>::unsafe_clear();
+
+		let bag_thresh = T::BagThresholds::get();
+		let low = bag_thresh[0];
+		let mid = bag_thresh[1];
+		let high = bag_thresh[2];
+
+		let rebag_budget = <T as Config<I>>::MaxAutoRebagPerBlock::get();
+		let n = rebag_budget + 5;
+
+		// Insert nodes with varying scores
+		for i in 0..n {
+			let node: T::AccountId = account("node", i, 0);
+			let score = match i % 3 {
+				0 => low - One::one(),
+				1 => mid - One::one(),
+				_ => high - One::one(),
+			};
+			assert_ok!(List::<T, _>::insert(node.clone(), score));
+		}
+
+		// Corrupt some nodes to simulate edge cases
+		for i in (0..n).step_by(4) {
+			let node: T::AccountId = account("node", i, 0);
+			let _ = List::<T, _>::remove(&node); // orphan nodes
+		}
+
+		// Now set new scores that will move nodes into higher bags
+		for i in 0..n {
+			let node: T::AccountId = account("node", i, 0);
+			let new_score = match i % 3 {
+				0 => mid,
+				1 => high,
+				_ => high + high, // force into a new top bag
+			};
+			T::ScoreProvider::set_score_of(&node, new_score);
+		}
+
+		// Ensure we have at least three bags populated before rebag
+		assert!(List::<T, _>::get_bags().len() >= 2);
+	}
+	: {
+		use frame_support::traits::Hooks;
+		<Pallet<T, I> as Hooks<_>>::on_idle(0u32.into(), Weight::MAX);
+	}
+	verify {
+		// Count how many nodes ended up in higher bags
+		let total_rebagged: usize = List::<T, _>::get_bags()
```

Comment:
won't counting the events here be a more accurate way to know how many you rebagged?